hoomd/snapshot.py

```diff
@@ -22,9 +22,13 @@ def box(self):
 
     @box.setter
     def box(self, box):
-        b = _hoomd.BoxDim(box[0], box[1], box[2])
-        b.setTiltFactors(box[3], box[4], box[5])
-        self._cpp_obj._global_box = b
+        if isinstance(box, hoomd.Box):
+            self._cpp_obj._global_box = hoomd.Box._from_cpp(
+                box._cpp_obj)._cpp_obj
+        else:
+            b = _hoomd.BoxDim(box[0], box[1], box[2])
+            b.setTiltFactors(box[3], box[4], box[5])
+            self._cpp_obj._global_box = b
```

Comment:
I know I've advised against duplicating freud logic before, but in this case I think it's potentially warranted: have you considered replicating freud's `from_box` method in hoomd.Box? Then you could call that here, and that method could handle more general cases than just a length-6 sequence (such as a dictionary or a freud.Box or anything else with the right attributes).

Reply:
I agree this would be the desired behavior, here and in `State.box = box`. I just haven't implemented the logic.